File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.py
def solution(n):
    fi_list = [0,1]
    for i in range(2,n+1):
        fi_list.append((fi_list[i-2]+fi_list[i-1])%1234567)
    return fi_list[n]

# def solution(n):
#     now_num = 0
#     next_num = 1
#     for i in range(0,n):
#         now_num, next_num = next_num, now_num + next_num
#     return now_num

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
import java.util.*;
class Solution {
    public int solution(int n) {
        ArrayList<Integer> fi_list = new ArrayList<>();
        fi_list.add(0);
        fi_list.add(1);

        for(int i = 2; i <= n; i++){
            fi_list.add((fi_list.get(i-2)+fi_list.get(i-1))%1234567);
        }
        return fi_list.get(n);
    }
}